### PR TITLE
Feature: Forward, Up, & Right vectors

### PR DESCRIPTION
**Pull request by wozniakty**
Transform now contains forward, up, and right functions to return the axes of the rotation quaternion.

### CODE REVIEW

File: source/core/gameobject.d [hunk 1]
```diff
@@ -425,6 +425,81 @@ public:
...
+     *
+     * Returns: The forward axis of the current transform
+     */
+    final @property const shared(vec3) forward()
+    {
+        return shared vec3( 2 * (rotation.x * rotation.z + rotation.w * rotation.y),
+                            2 * (rotation.y * rotation.x - rotation.w * rotation.x),
+                            1 - 2 * (rotation.x * rotation.x + rotation.y * rotation.y ));
+    }
+    ///
```
Comment (dislogical) [inline source/core/gameobject.d:442]: For those curious, this adds the unittest as an example in the docs for `forward`.

**Comment by dislogical** [comment]
:a:

**Comment by ericchristenson** [comment]
:100:

**Comment by brooklynlittell** [comment]
:+1:

**Pull request merged**